mailpoet/lib/Automation/Engine/Builder/UpdateWorkflowController.php

```diff
@@ -27,6 +27,12 @@ public function updateWorkflow(int $id, array $data): Workflow {
       throw Exceptions::workflowNotFound($id);
     }
 
+    if (array_key_exists('name', $data)) {
+      $this->checkWorkflowName($data['name']);
+      $workflow->setName($data['name']);
+      $this->storage->updateWorkflow($workflow);
+    }
```

Comment:
Now that we have two IFs here, we should probably extract `$this->storage->updateWorkflow($workflow);` below to run the query only once even in case both fields are updated.

Reply:
Fixed.